zio-http/shared/src/main/scala/zio/http/Server.scala

```diff
@@ -331,11 +336,18 @@ object Server extends ServerPlatformSpecific {
 
   def serve[R](
     httpApp: HttpApp[R],
+    logStartingServer: Boolean = true,
   )(implicit trace: Trace): URIO[R with Server, Nothing] =
     install(httpApp) *> ZIO.never
 
-  def install[R](httpApp: HttpApp[R])(implicit trace: Trace): URIO[R with Server, Int] = {
-    ZIO.serviceWithZIO[Server](_.install(httpApp)) *> ZIO.service[Server].map(_.port)
+  def install[R](httpApp: HttpApp[R], logStartingServer: Boolean = true)(implicit
+    trace: Trace,
+  ): URIO[R with Server, Int] = {
+    ZIO.serviceWithZIO[Server] { server =>
+      val logMessage = ZIO.when(logStartingServer)(ZIO.logInfo(s"ZIO HTTP server is running at port ${server.port}"))
+
+      logMessage *> server.install(httpApp).tap(_ => logMessage).as(0)
```

Comment:
`install` should return port on which server is running, but in your case it will always return 0, that's not right. Plus see my comment about putting log into `serve`